File: src/models/UserStory.ts
import {Entity,PrimaryGeneratedColumn,Column,ManyToOne,JoinColumn} from 'typeorm';
import { User } from './User';
import { Story } from './Story';

@Entity('user_story')
export class UserStory {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column()
  user_id!: number;

  @Column()
  story_id!: number;

  @Column({ type: 'varchar' })
  role!: string; // autor, editor, leitor, tradutor...

  @ManyToOne(() => User, (user) => user.userStories)
  @JoinColumn({ name: 'user_id' })
  user!: User;

  @ManyToOne(() => Story, (story) => story.userStories)
  @JoinColumn({ name: 'story_id' })
  story!: Story;
}

File: backend/src/models/user_story.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from src.db.database import Base


class UserStory(Base):
    __tablename__ = "user_story"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    story_id = Column(Integer, ForeignKey("story.id"), nullable=False)
    role = Column(String, nullable=False)  # autor, editor, leitor, tradutor...

    user = relationship("User", back_populates="user_stories")
    story = relationship("Story", back_populates="user_stories")
